File: OpenTokFs.ResponseTypes/OpenTokArchive.cs
using System;

namespace OpenTokFs.ResponseTypes {
    public class OpenTokArchive {
        public long CreatedAt { get; set; } = 0L;
        public int Duration { get; set; } = 0;
        public bool HasAudio { get; set; } = false;
        public bool HasVideo { get; set; } = false;
        public string Id { get; set; } = "";
        public string? Name { get; set; } = null;
        public string? OutputMode { get; set; } = null;
        public int ProjectId { get; set; } = 0;
        public string Reason { get; set; } = "";
        public string? Resolution { get; set; } = null;
        public string SessionId { get; set; } = "";
        public long Size { get; set; } = 0;
        public string Status { get; set; } = "";
        public string? Url { get; set; } = null;

        public DateTimeOffset GetCreationTime() => DateTimeOffset.FromUnixTimeMilliseconds(CreatedAt);
        public TimeSpan GetDuration() => TimeSpan.FromSeconds(Duration);

        public override string ToString() => $"{Id} ({Name ?? "no name"}) ({OutputMode}, {Status})";
    }
}

File: OpenTokFs/ResponseTypes/OpenTokArchive.fs
namespace OpenTokFs.ResponseTypes

open System

type OpenTokArchive = {
    createdAt: int64
    duration: int
    hasAudio: bool
    hasVideo: bool
    id: string
    name: string // or null
    outputMode: string // or null
    projectId: int
    reason: string
    resolution: string // or null
    sessionId: string
    size: int64
    status: string
    url: string // or null
} with
    member this.GetCreationTime () = DateTimeOffset.FromUnixTimeMilliseconds this.createdAt
    member this.GetDuration () = TimeSpan.FromSeconds this.duration
    override this.ToString () = String.concat " " [
        this.id
        if not (isNull this.name) then $"({this.name})"
        $"({this.outputMode}, {this.status})"
    ]